File: src/backend/mdfe/src/model/mdfe_informacao_seguro_model.py
from src import db


class MdfeInformacaoSeguroModel(db.Model):
    __tablename__ = 'mdfe_informacao_seguro'

    id = db.Column(db.Integer, primary_key=True)
    responsavel = db.Column(db.Integer)
    cnpj_cpf = db.Column(db.String(14))
    seguradora = db.Column(db.String(11))
    cnpj_seguradora = db.Column(db.String(14))
    apolice = db.Column(db.String(20))
    averbacao = db.Column(db.String(40))
    id_mdfe_cabecalho = db.Column(db.Integer, db.ForeignKey('mdfe_cabecalho.id'))


    def mapping(self, data):
        self.id = data.get('id')
        self.id_mdfe_cabecalho = data.get('idMdfeCabecalho')
        self.responsavel = data.get('responsavel')
        self.cnpj_cpf = data.get('cnpjCpf')
        self.seguradora = data.get('seguradora')
        self.cnpj_seguradora = data.get('cnpjSeguradora')
        self.apolice = data.get('apolice')
        self.averbacao = data.get('averbacao')

    def serialize(self):
        return {
            'id': self.id,
            'idMdfeCabecalho': self.id_mdfe_cabecalho,
            'responsavel': self.responsavel,
            'cnpjCpf': self.cnpj_cpf,
            'seguradora': self.seguradora,
            'cnpjSeguradora': self.cnpj_seguradora,
            'apolice': self.apolice,
            'averbacao': self.averbacao,
        }

File: src/frontend/mdfe/lib/app/data/model/mdfe_informacao_seguro_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';

import 'package:mdfe/app/infra/infra_imports.dart';

class MdfeInformacaoSeguroModel {
	int? id;
	int? idMdfeCabecalho;
	int? responsavel;
	String? cnpjCpf;
	String? seguradora;
	String? cnpjSeguradora;
	String? apolice;
	String? averbacao;

	MdfeInformacaoSeguroModel({
		this.id,
		this.idMdfeCabecalho,
		this.responsavel,
		this.cnpjCpf,
		this.seguradora,
		this.cnpjSeguradora,
		this.apolice,
		this.averbacao,
	});

	static List<String> dbColumns = <String>[
		'id',
		'responsavel',
		'cnpj_cpf',
		'seguradora',
		'cnpj_seguradora',
		'apolice',
		'averbacao',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Responsavel',
		'Cnpj Cpf',
		'Seguradora',
		'Cnpj Seguradora',
		'Apolice',
		'Averbacao',
	];

	MdfeInformacaoSeguroModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		idMdfeCabecalho = jsonData['idMdfeCabecalho'];
		responsavel = jsonData['responsavel'];
		cnpjCpf = jsonData['cnpjCpf'];
		seguradora = jsonData['seguradora'];
		cnpjSeguradora = jsonData['cnpjSeguradora'];
		apolice = jsonData['apolice'];
		averbacao = jsonData['averbacao'];
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['idMdfeCabecalho'] = idMdfeCabecalho != 0 ? idMdfeCabecalho : null;
		jsonData['responsavel'] = responsavel;
		jsonData['cnpjCpf'] = Util.removeMask(cnpjCpf);
		jsonData['seguradora'] = seguradora;
		jsonData['cnpjSeguradora'] = Util.removeMask(cnpjSeguradora);
		jsonData['apolice'] = apolice;
		jsonData['averbacao'] = averbacao;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		idMdfeCabecalho = plutoRow.cells['idMdfeCabecalho']?.value;
		responsavel = plutoRow.cells['responsavel']?.value;
		cnpjCpf = plutoRow.cells['cnpjCpf']?.value;
		seguradora = plutoRow.cells['seguradora']?.value;
		cnpjSeguradora = plutoRow.cells['cnpjSeguradora']?.value;
		apolice = plutoRow.cells['apolice']?.value;
		averbacao = plutoRow.cells['averbacao']?.value;
	}	

	MdfeInformacaoSeguroModel clone() {
		return MdfeInformacaoSeguroModel(
			id: id,
			idMdfeCabecalho: idMdfeCabecalho,
			responsavel: responsavel,
			cnpjCpf: cnpjCpf,
			seguradora: seguradora,
			cnpjSeguradora: cnpjSeguradora,
			apolice: apolice,
			averbacao: averbacao,
		);			
	}

	
}